fn.go

```diff
@@ -20,34 +31,182 @@ type Function struct {
 	log logging.Logger
 }
 
+const (
+	invalidFunctionFmt = "invalid function input: %s"
+	wrongTempErr       = "templates are required either inline or from filesystem path"
+	cannotGetErr       = "cannot get the function cd"
+	cannotParseErr     = "cannot parse the provided templates"
+)
+
+const (
+	AnnotationKeyCompositionResourceName = "crossplane.io/composition-resource-name"
+)
+
 // RunFunction runs the Function.
 func (f *Function) RunFunction(_ context.Context, req *fnv1beta1.RunFunctionRequest) (*fnv1beta1.RunFunctionResponse, error) {
 	f.log.Info("Running Function", "tag", req.GetMeta().GetTag())
 
-	// This creates a new response to the supplied request. Note that Functions
-	// are run in a pipeline! Other Functions may have run before this one. If
-	// they did, response.To will copy their desired state from req to rsp. Be
-	// sure to pass through any desired state your Function is not concerned
-	// with unmodified.
 	rsp := response.To(req, response.DefaultTTL)
 
-	// Input is supplied by the author of a Composition when they choose to run
-	// your Function. Input is arbitrary, except that it must be a KRM-like
-	// object. Supporting input is also optional - if you don't need to you can
-	// delete this, and delete the input directory.
 	in := &v1beta1.Input{}
 	if err := request.GetInput(req, in); err != nil {
-		response.Fatal(rsp, errors.Wrapf(err, "cannot get Function input from %T", req))
+		response.Fatal(rsp, errors.Wrapf(err, "cannot get Function cd from %T", req))
+		return rsp, nil
+	}
+
+	if !isValidInputSource(in) {
+		response.Fatal(rsp, errors.New(fmt.Sprintf(invalidFunctionFmt, wrongTempErr)))
+		return rsp, nil
+	}
+
+	tg, err := NewTemplateSourceGetter(in)
+	if err != nil {
+		response.Fatal(rsp, errors.Wrap(err, fmt.Sprintf(invalidFunctionFmt, cannotGetErr)))
+		return rsp, nil
+	}
+
+	tmpl, err := GetNewTemplateWithFunctionMaps().Parse(tg.GetTemplate())
+	if err != nil {
+		response.Fatal(rsp, errors.Wrap(err, fmt.Sprintf(invalidFunctionFmt, cannotParseErr)))
+		return rsp, nil
+	}
+
+	reqMap, err := convertToMap(req)
+	if err != nil {
+		response.Fatal(rsp, errors.Wrap(err, "cannot convert request to map"))
+		return rsp, nil
+	}
+
+	f.log.Debug("constructed request map", "request", reqMap)
+
+	buf := &bytes.Buffer{}
+
+	if err := tmpl.Execute(buf, reqMap); err != nil {
+		response.Fatal(rsp, errors.Wrap(err, "cannot execute template"))
+		return rsp, nil
+	}
+
+	f.log.Debug("rendered manifests", "manifests", buf.String())
+
+	// Parse the rendered manifests.
+	var objs []*unstructured.Unstructured
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewBufferString(buf.String()), 1024)
```

Comment:
What does this `1024` mean? Does it work like a limit meaning we cannot have yamls bigger than this or just a buffer size for optimization?